hera_sim/foregrounds.py

```diff
@@ -7,61 +7,48 @@
+                       fringe_filter_type='tophat', **fringe_filter_kwargs):
     """
-    Produce visibilities containing diffuse foregrounds.
+    Model diffuse foreground visibility.
 
     Args:
-        Tsky (float): sky temperature [in mK?]
+        Tsky_mdl (RectBivariateSpline): Model of total autocorrelation power, see noise.HERA_Tsky_mdl
         lsts (ndarray): LSTs [radians]
-        fqs (ndarray): frequencies [GHz]
-        bl_len_ns (float): East-West baseline length [nanosec]
-        bm_poly (ndarray): a polynomial for beam size with frequency.
-        scalar (float): WHAT'S THIS?
-        fr_width (float): width of Gaussian FR filter in 1 / sec
-        fr_max_mult (float): multiplier of fr_max to get lst_grid resolution
+        fqs (ndarray): Frequencies [GHz]
+        bl_vec (ndarray): East-North-Up (i.e. Topocentric) baseline vector in nanoseconds [East, North, Up]
+        bm_poly (ndarray): beam scalar polynomial object, see noise.HERA_BEAM_POLY
+        standoff (float): baseline horizon buffer [ns] for modeling suprahorizon emission
+        delay_filter_type (str): type of delay filter to use, see utils.gen_delay_filter()
+        fringe_filter_type (str): type of fringe-rate filter, see utils.gen_fringe_filter()
+        fringe_filter_kwargs: kwargs given fringe_filter_type, see utils.gen_fringe_filter()
 
     Returns:
-        2D ndarray : visibilities at each lst, fq pair.
+        data (ndarray): diffuse foreground visibility
     """
-    # If an auto-correlation, return the beam-weighted integrated sky.
-    if utils.get_bl_len_magnitude(bl_len_ns) == 0:
-        return Tsky(lsts, fqs) / noise.jy2T(fqs, bm_poly=bm_poly)
-
-    # Get the maximum fringe rate corresponding to a time scale over
-    # which co-ordinates pass through the beam.
-    beam_widths = np.polyval(bm_poly, fqs)
-    fr_max_beam = np.max(2*np.pi/(aipy.const.sidereal_day * beam_widths))
-    fr_max = np.max(utils.calc_max_fringe_rate(fqs, bl_len_ns))
+    # generate a Tsky visibility in time and freq space, convert from K to Jy
+    Tsky = Tsky_mdl(lsts, fqs)
+    data = np.asarray(Tsky * 1e3 / noise.jy2T(fqs, bm_poly=bm_poly), np.complex)
 
-    fr_max = max(fr_max, fr_max_beam)
+    # multiply by white noise
+    data *= noise.white_noise((len(lsts), len(fqs)))
 
-    dt = 1.0 / (fr_max_mult * fr_max)  # over-resolve by fr_mult factor
-    ntimes = int(np.around(aipy.const.sidereal_day / dt))
+    # fringe rate filter across time using projected East-West distance
+    data, fringe_filter = utils.rough_fringe_filter(data, lsts, fqs, np.abs(bl_vec[0]), filter_type=fringe_filter_type, **fringe_filter_kwargs)
 
-    lst_grid = np.linspace(0, 2 * np.pi, ntimes, endpoint=False)
-    nos = Tsky(lst_grid, fqs) * noise.white_noise((ntimes, fqs.size))
+    # delay filter across freq
+    data, delay_filter = utils.rough_delay_filter(data, fqs, np.linalg.norm(bl_vec), standoff=standoff, filter_type=delay_filter_type)
```

Comment:
will this work for auto-correlations?

Reply:
I don't think so, because the visibility will still be complex. the auto-correlation should actually just be `Tsky` in this case..